plugin/validator_test.go

```diff
@@ -27,6 +27,47 @@ type KongPluginSchema struct {
 	Fields []map[string]interface{} `json:"fields,omitempty" yaml:"fields,omitempty"`
 }
 
+func TestValidator_ValidateSchema(t *testing.T) {
+	v, err := NewValidator(ValidatorOpts{})
+	assert.Nil(t, err)
+
+	t.Run("validate a valid schema", func(t *testing.T) {
+		schema, err := ioutil.ReadFile("testdata/key-auth.lua")
+		assert.Nil(t, err)
+		pluginName, err := v.ValidateSchema(string(schema))
+		assert.Nil(t, err)
+		assert.EqualValues(t, "key-auth", pluginName)
+	})
```

Comment:
Could you add a test "re-validating different schemas for the same name works"?
I want to ensure that no global state is mutated by this line:
```
   ok, err = Entity.new_subschema(validate_plugins, plugin_name, plugin_schema)
```
in the code.